sys/sysent.c

```diff
@@ -24,8 +28,69 @@ int sys_write(thread_t *td, syscall_args_t *args) {
   return -EBADF;
 }
 
+/* This is just a stub. A full implementation of this syscall will probably
+   deserve a separate file. */
+/* Note that this sbrk implementation does not actually extend .data section,
+   because we have no guarantee that there is any free space after .data in the
+   memory map. But it does not matter much, because no application would assume
+   that we are actually expanding .data, it will use the pointer returned by
+   sbrk. */
+int sys_sbrk(thread_t *td, syscall_args_t *args) {
+  size_t increment = (size_t)args->args[0];
+
+  kprintf("[syscall] sbrk(%zu)\n", increment);
+
+  assert(td->td_uspace);
+
+  if (!td->td_uspace->td_brk_entry) {
```

Comment:
For the record, FreeBSD [does not implement `sbrk` at all](http://bxr.su/FreeBSD/sys/vm/vm_mmap.c#121).

IIRC historically, back when expanding the data segment was the only way of allocating memory, data segment and process stack were placed in the same segment, so a movable pointer *break* was used to choose the point that divided these two. So I suppose what *brk(2)* says is mostly a historical convention. The test you did is probably a valid proof that applications don't generally require `sbrk` to expand `.data`, because if this was a case, such programs would not function correctly on Linux.